### ISSUE TITLE
Ubuntu  24.04 next release has no libmpv1 but libmpv2

### ISSUE THREAD

**Issue by stdevPavelmc** [Stremio/stremio-shell#393]
System: Linux Ubuntu 24.04 LTS (new develop branch)
Error:  libmpv1 not exist, they migrated to libmpv2

![image](https://github.com/Stremio/stremio-shell/assets/42220093/523cb776-1453-42d0-91a0-691b4e42b487)

**Comment by yusi1** [comment]
Same issue on Debian 12 Bookworm:
```
 ~/Downloads $ sudo dpkg -i stremio_4.4.168-1_amd64.deb
[sudo] password for yaslam:
Selecting previously unselected package stremio.
(Reading database ... 390153 files and directories currently installed.)
Preparing to unpack stremio_4.4.168-1_amd64.deb ...
Unpacking stremio (4.4.168-1) ...
dpkg: dependency problems prevent configuration of stremio:
 stremio depends on nodejs; however:
  Package nodejs is not installed.
 stremio depends on libmpv1 (>= 0.30.0); however:
  Package libmpv1 is not installed.
 stremio depends on qml-module-qtwebchannel (>= 5.9.5); however:
  Package qml-module-qtwebchannel is not installed.
 stremio depends on libfdk-aac2 (>= 2.0.2); however:
  Package libfdk-aac2 is not installed.

dpkg: error processing package stremio (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 stremio
 ✘ ~/Downloads $ sudo apt install nodejs libmpv1 qml-module-qtwebchannel libfdk-aac2
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Package libmpv1 is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'libmpv1' has no installation candidate
 ✘ ~/Downloads $

```

**Comment by laserburn** [comment]
Yup, definitely a problem. Now that new Ubuntu LTS is officially out, many will flock to it and discover that Stremio can't be installed. Must I polute the new Ubuntu 24.04 with flatpaks already? :(

**Comment by gustavo-dotcom** [comment]
Same issue here. Maybe launch Stremio as snap for Ubuntu?

**Comment by midhun3301** [comment]
Same issue. Latest Ubuntu 24.04 LTS.

**Comment by Corbachini33** [comment]
> Same issue here. Maybe launch Stremio as snap for Ubuntu? 

I've been using the flatpak until the deb is ready and its been working fine.

**Comment by midhun3301** [comment]
> I've been using the flatpak until the deb is ready and its been working fine.

Do you have a Nvidia GPU? Is Stremio able to use to for video decoding?

**Comment by Corbachini33** [comment]
> > I've been using the flatpak until the deb is ready and its been working fine.
> 
> Do you have a Nvidia GPU? Is Stremio able to use to for video decoding?

Haven't tried because the CPU I have is good enough for 4k. I know there were problems with NVIDIA and video decoding on Linux but I don't know if they also affect Stremio.

**Comment by richardiago** [comment]
Any updates on this one?

**Comment by kpjim** [comment]
As a workaround I was able to repackage the .deb, changing the dependencies in the process:
`dpkg-deb -R stremio.deb tmpdir`
edit the tmpdir/DEBIAN/control file to change the libmpv dependeny to: libmpv2 (ubuntu should already have this package).

At this point you also need to edit the stremio binary with hexedit (or a similar tool) so that you replace the loading of libmpv.so.1 with libmpv.so.2:
`hexedit tmpdir/opt/stremio/stremio`
Now search the hex string: '6C 69 62 6D 70 76 2E 73 6F 2E 31' and replace the 31 with 32. Save, exit and then repackage:
`dpkg-deb -b tmpdir stremio_dirty.deb`

Install
`sudo dpkg -i stremio_dirty.deb.`
Worked for me

**Comment by Kabir5296** [comment]
Run this bash script to simply install stremio without the libmpv dependency issue

[https://github.com/Kabir5296/Stremio-Ubuntu24.04-Fix](https://github.com/Kabir5296/Stremio-Ubuntu24.04-Fix)

**Comment by angeloscle** [comment]
```
sudo apt install flatpak
sudo flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
flatpak update
flatpak install flathub com.stremio.Stremio
```

**Comment by vejeta** [comment]
If you're having compilation issues, there's now a pre-built option available!

See PR #455  - I've added an APT repository that provides ready-to-use packages, eliminating the need to compile from source:
```bash
wget -qO - https://debian.vejeta.com/key.gpg | sudo gpg --dearmor -o /usr/share/keyrings/stremio-debian.gpg
echo "deb [signed-by=/usr/share/keyrings/stremio-debian.gpg] https://debian.vejeta.com trixie main non-free" | sudo tee /etc/apt/sources.list.d/stremio.list
sudo apt update
sudo apt install stremio stremio-server
```

I build this to be included in the official Debian repositories. I have not tested it on Ubuntu, let me know how it goes, in case a build for Ubuntu needs to be made.

This should work out-of-the-box without any Qt5/build dependency issues. I have also updated the DEBIAN.md instructions.

**Comment by Cruel** [comment]
Great, now latest Ubuntu uses `librubberband3` so that has to be patched too...

**Comment by JuanClaude** [comment]
Same on Xubuntu 24.04 , thanks for this post. I'll install it on flatpak untill there's an easy fix by the official website.

**Comment by oussama-m** [comment]
I forked @Kabir5296 repo and fixed the issue for librubberband2.
It's the same type of issue as libmpv1.
https://github.com/oussama-m/Stremio-Ubuntu24-26.04-Fix

Basically it just add these line to the script to replace the dependency from librubberband2 to librubberband3:
```
# Modify the dependency from librubberband2 to librubberband3
sed -i 's/librubberband2/librubberband3/g' stremio_extract/DEBIAN/control
sed -i 's/\x6C\x69\x62\x72\x75\x62\x62\x65\x72\x62\x61\x6E\x64\x2E\x73\x6F\x2E\x32/\x6C\x69\x62\x72\x75\x62\x62\x65\x72\x62\x61\x6E\x64\x2E\x73\x6F\x2E\x33/g' stremio_extract/opt/stremio/stremio
```
